ts/features/idpay/onboarding/xstate/machine.ts

```diff
@@ -387,6 +411,10 @@ const createIDPayOnboardingMachine = () =>
               actions: "navigateToInitiativeMonitoringScreen"
             }
           }
+        },
+
+        SESSION_EXPIRED: {
+          entry: ["handleSessionExpired", "exitOnboarding"]
```

Comment:
correct me if I'm wrong, but doesn't this kill the flow in case of deauth?

If that's the case, since the onboarding flow can potentially take quite a bit, and even though I'm pretty sure you've tried this already, wouldn't it be better to store the actual data somewhere in the global state and restore the machine after the sessionExpired handling process? We could add the stashing to the `entry` and make  `handleSessionExpired` handle the post-auth navigation back in the machine -- at least I think --

I believe users would lash back quite a bit if they lost all their progress after the possibly lengthy multi criteria screen

Reply:
We have currently decided not to reset progress after a session has expired. We may improve this in a future update.
Anyway, I don't think that saving the state elsewhere to restore it later would be the right (and easiest) solution. Instead, we should let the machine wait for the session to be refreshed.